### DISCUSSION TITLE
Create the same file differently, depending on a condition?

### DISCUSSION THREAD

**Discussion by FranklinYu** [twpayne/chezmoi#2277] [Q&A] [answered]
I’d like `~/.cache` to be a normal directory on Linux, but a symbolic link on Darwin. Right now I’m using a script. Is there any idiomatic way?

I know how to create a conditional symbolic link with templates. I think the problem is how to create the directory conditionally. If I understand it correctly, creating an empty directory is like

```plain
- dot_cache
   `- .keep
```

Even I make `.keep` empty, the directory will still be created.

**Comment by twpayne** [comment, marked as answer]
Currently, there is no way for the same path (`~/.cache`) to have different types on different machines. However, there is a work-around for this specific case.

`~/.cache` is a directory on Linux by default, so you can ignore it on Linux, and then make it a symlink on macOS. For example (not tested):

In your `.chezmoiignore`:

```
{{ if ne .chezmoi.os "darwin" }}
.cache
{{ end }}
```

in `~/.local/share/chezmoi/symlink_dot_cache`:

```
Library/Caches
```

This will make `~/.cache` a symlink to `~/Library/Caches` on macOS and ignore `~/.cache` on all other OSes.

**Reply by FranklinYu** [reply]
If I recall correctly, `~/.cache/` directory is missing on a fresh-install of Debian. This is problematic because many applications would create the directory `~/.cache/foobar` only when `~/.cache/` exists. My point is to create this missing `~/.cache/` on Linux, not to ignore it. From your answer, I infer that a `run_` script is best we have so far. Thanks anyway!

**Reply by twpayne** [reply]
Yeah, a `run_` script is the current best option here.

**Reply by FranklinYu** [reply]
By the way, since a `run_` script is opaque to Chezmoi, is there any feature/FR for the script to “tell” Chezmoi that it is managing some file/directory? This way, the file/directory (target) can be shown in the `managed` sub-command, and Chezmoi can warn me when the target is changed outside of Chezmoi.

**Reply by twpayne** [reply]
No, there is no way to tell chezmoi which files and directories are touched by scripts.
